File: RedSheet.DbEntities/Models/vValueObjective.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vValueObjectives", Schema = "dbo")]
	[Serializable]
	public partial class vValueObjective  
	{
 
	[Required] 
	[MaxLength(200)]
	public string ClassName { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ValueObjectiveId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string ValueObjectiveName { get; set; } 

	public vValueObjective()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-value-objective.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vValueObjective {
    constructor(vValueObjective?: vValueObjective )  {
        let properties = [ "className", "valueObjectiveId", "valueObjectiveName",];
        for (let property of properties)
            if (vValueObjective && vValueObjective[property])
                this[property] = vValueObjective[property];
    }
 
    @required()
    @maxLength(200)
	className : string =   undefined;
 
	valueObjectiveId : number =   0 ;
 
    @required()
    @maxLength(200)
	valueObjectiveName : string =   undefined;


}
